### DISCUSSION TITLE
The problem of pn.regenerate_models()

### DISCUSSION THREAD

**Discussion by Xknightfyy** [PMEAL/OpenPNM#2733] [Q&A]
Dear Professor Jeff Gostick,

Sorry to bother you again, but I can't solve this problem, hope to get your help. 1. When I use the built-in function to generate the pore distribution, I can get the correct result, but when I use pn.regenerate_models(), its pore diameter will become the original model, which will cause me an error.
![1681388793491](https://user-images.githubusercontent.com/120148734/231757794-06a127c9-78e7-4418-b146-543f4f6bd689.png)

2. When I use the stitched function and then run the drainage algorithm, I get the same intrusion pressure, how does this happen?
![1681388828375](https://user-images.githubusercontent.com/120148734/231757924-39789e26-8bb9-45c1-95f8-93e611d78022.png)
Hope to get your reply soon, best wishes. xu

**Comment by jgostick** [comment]
The first problem is because when you do ``pn.add_model_collection`` on line 35 it is overwriting the previous model which you added on line 29.  

The second question is not detailed enough for me to answer.  Could you open a separate discussion and state your question in a bit more detail?

**Reply by Xknightfyy** [reply]
Hello, thank you for your answer!!! But I still have a question of how should I determine the termination condition in the InvasionPercolation algorithm, such as the intrusion process terminates when the intrusion reaches the top hole, and I am now experiencing some difficulties here.It's nice to have a short code example, thanks again, sorry to bother you.
